File: T2Lc/Compiler.cs
using Antlr4.Runtime;
using Antlr4.Runtime.Tree;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using T2Lc.Antlr;

namespace T2Lc
{
    public class Compiler
    {
        public Compiler(string code)
        {
            ICharStream stream = CharStreams.fromString(code);
            ITokenSource lexer = new t2lLexer(stream);
            ITokenStream tokens = new CommonTokenStream(lexer);

            t2lParser parser = new(tokens);
            parser.BuildParseTree = true;
            IParseTree tree = parser.start();

            Listener listener = new();
            ParseTreeWalker.Default.Walk(listener, tree);
        }
    }
}

File: T2Lc/Compiler.cpp
#include "Compiler.h"

#include <sstream>

#include "antlr4/t2lLexer.h"
#include "antlr4/t2lParser.h"

#include "Visitor.h"

Compiler::Compiler(string code)
{
	stringstream stream(code);

	ANTLRInputStream input(stream);
	t2lLexer lexer(&input);
	CommonTokenStream tokens(&lexer);
	t2lParser parser(&tokens);

	tree::ParseTree* tree = parser.start();
	Visitor visitor;
	visitor.visit(tree);
}
